src/v/storage/disk_log_impl.cc

```diff
@@ -2326,10 +2330,35 @@ int64_t disk_log_impl::compaction_backlog() const {
     // or enabling compaction on a previously non-compacted topic.
     static constexpr size_t limit_segments_this_term = 1024;
 
-    for (auto& s : _segs) {
-        if (!s->finished_self_compaction()) {
-            backlog += static_cast<int64_t>(s->size_bytes());
+    if (config::shard_local_cfg().log_compaction_use_sliding_window()) {
+        ss::abort_source never_abort;
+        // NOTE: the backlog estimate is just that -- an estimate. Don't bother
+        // passing in real offsets, priority classes, etc.
+        auto segs = find_sliding_range(compaction_config{
+          model::offset::max(), ss::default_priority_class(), never_abort});
+        for (auto& s : segs) {
+            if (!s->finished_self_compaction()) {
+                backlog += static_cast<int64_t>(s->size_bytes());
+            }
+            // TODO: using the average compaction ratio here is a crude
+            // approximation, since the ratio also includes the size reduction
+            // from windowed compaction itself.
+            if (!s->finished_windowed_compaction()) {
```

Comment:
Should remove the `if()` here, since all segments in the sliding range will be rewritten* (*except segments that have only one record).

Also I think the `* cf` should only be applied to segments that aren't self compacted yet.